### DISCUSSION TITLE
How to use JSON mode with OpenAI

### DISCUSSION THREAD

**Discussion by khiconit** [run-llama/LlamaIndexTS#1586] [Q&A]
Open AI and Ollama now support output format. Please support it soon

Documentation: https://platform.openai.com/docs/guides/structured-outputs?context=without_parse#json-mode

**Comment by marcusschiesser** [comment]
JSON mode is already supported via:
```
const llm = new OpenAI({
    model: "gpt-4o",
    additionalChatOptions: { response_format: { type: "json_object" } },
  });
```

but we should generally support structured output

**Reply by AVtheking** [reply]
> ```
> 
> ```
> 
> 
>     
>   
> 
> but we should generally support structured output

Do you mean having a parameter for the structured output with zod schema ?

**Reply by marcusschiesser** [reply]
yes, and generally support non openai LLMs

**Reply by AVtheking** [reply]
but what about those llm which are not having support for structured output ? do you mean to add it as a optional field with a flag just like we have for tool call for the llm ?

**Reply by AVtheking** [reply]
Created a PR for this https://github.com/run-llama/LlamaIndexTS/pull/1749 , let me know if the approach is correct

**Comment by marcusschiesser** [comment]
just saw we have an issue for structured output already: https://github.com/run-llama/LlamaIndexTS/issues/1249 - converting this to a discussion
